### ISSUE TITLE
State of Slush

### ISSUE THREAD

**Issue by kokujin** [slushjs/slush#51]
I wonder what the state of Slush is at the moment. 3665 generators compared to 285+. Slush was(is?) supposed to be a replacement for Yeoman, but it seems as if the development or use is slowing down. Should there be a list of approved generators to change this situation?

**Comment by joakimbeng** [comment]
Yes, Slush _can_ be a replacement for Yeoman, but as you say there are far more generators for Yeoman than for Slush. And if you look at it from the user's perspective there's hardly any difference between a Yeoman or Slush generator, so as a user you're probably going with the scaffolding system with the most generators.

Slush was created to ease the development of generators compared to how to develop them for Yeoman. At the time of Slush's release IMO it did in fact make that easier. But Yeoman may have evolved since then so the developer experience may be much better today.

In hindsight Slush wasn't that much better than Yeoman for developers, to consider switching scaffolding system. So I think the developers bets on the system with the most generators as well.

For me the problem with all generators are maintaining them, because of the fast pace in the JS community. Best practices and libs are changing almost constantly so it's hard to make generic and long lasting generators. This is also related to #52 which I see [Yeoman has a solution for](http://yeoman.io/authoring/dependencies.html). The key to easing the developer experience regarding maintenance of generators is composability and resusability which [Yeoman also seems to have come a long way with](http://yeoman.io/authoring/composability.html).

**Conclusion:** I understand that both users and developers use the system with the most generators (Yeoman), so the development of Slush generators is slowing down. And I don't actually think a list of approved generators would change that situation. But I'm up for adding new contributors/maintainers to the Slush community with fresh ideas of where to take Slush! (wanna be a member? don't hesitate to ask)

---

**Side note:** I would love to see a new kind of scaffolding system which promotes reusability in its core, pure functions, no `this`, no inheritance, plain objects and perhaps promises. With a scaffolding system like that, one can really embrace the whole NPM ecosystem to build generators. My intention is to build something in that manner but, as always, haven't had the time.

**Comment by sameeri** [comment]
@joakimbeng 

Hope you are doing well. It has been sometime since i have last visited slush's repo and i stepped into look if slush was going to support gulp (version 4).

Anyway, i ran into this issue and looked at your thoughts and just wanted to pour mine out.

* Firstly thank you for creating slush. 
* slush, yeoman and any other scaffolding system tries to solve a core problem. So i feel that it is as you accurately describe - a **choice**. And since a lot of community revolves around yeoman, naturally there is a tendency for people to opt for a certain system than another.
* What is the purpose of their use describes whether they would use one system or another. 
* If a team/individual wants to get started quickly on something, say Angular, for instance, they would then look for a generator. They would look for opinions on how to organize and think about their application in that particular area of interest. 
* If a team/individual want to develop their own generators, things get interesting. Now, questions arise. How to think and how to work in this system. This choice has a lot of implications on what is available for them to learn and quickly apply.
* For example, slush is dependent on gulp. Users of gulp might quickly adopt and use the system. Maybe that was your original idea.
* Yeoman has its own way of thinking. They have a good set of docs and many examples (in the form of generators) at their core and a lot of people always thinking about the system and improving it.

A history of how i got started with slush.

* Team was using Yeoman (circa. 2013-14). I was on the .NET stack and had no clue of what was going on the JS team. They started out and were facilitating the integration of grunt + yeoman. They were trying to incorporate their own thought processes as our own generator.
* gulp comes, and team has had great relationship with the gulp guys
* It made sense to use code driven build process with gulp instead of  configuration oriented grunt.
* So our architect & team chose gulp. 
* I joined the JS team later.
* Our generators are still in yeoman. A lot of issues. People can't get back to that as they have to work on other important things.
* And slush comes in.
* There is a talk we should use slush instead of Yeoman.
* I get a sense of what's happening on the JS projects and try to learn.
* Scaffolding and how the other people are thinking about systems doesn't make sense.
* I try to learn yeoman, slush.
* I see slush as a very simple system to write generators in.
* I take the existing generators and move them to slush. I don't like how the old ones are written. So the slush ones are a complete rewrite.
* Now the original problem of many issues and only a couple of people knowing what this is, what its purpose is and how we should think about our systems makes me study slush code.
* slush code is super simple. It makes me happy.
* I try to document my thought processes using https://github.com/sameeri/Code-slush/wiki so that people know how to use and work with slush.
* I plan on writing a couple of simple generators to help establish this thought process.
* And i am at that point of time pretty sure that i want to create "good" generators in some systems i got started working on such as hapi.

And then i got busy with other stuff. :) But i consider slush a project that has helped me a lot. And i appreciate the thought process of making a simple system using another simple system.

These days, rapid learning, and working with anything new on the table that offers a new thinking is not too uncommon and to establish a good scaffolding system needs time and practice and vision.

And people like Dan Abramov throwing challenges to the community prompts people to think and build systems like https://github.com/rocjs/roc

Your thoughts and ideas are very interesting. And i would love to learn from you your ways of thinking.
